packages/vhd-lib/Vhd/VhdDirectory.js

```diff
@@ -285,7 +293,7 @@ exports.VhdDirectory = class VhdDirectory extends VhdAbstract {
   }
 
   async writeEntireBlock(block) {
-    await this._writeChunk(this.#getBlockPath(block.id), block.buffer)
+    await this._writeChunk(this.#getBlockPath(block.id), block.buffer, this.#dedup)
```

Comment:
should use the right flag to ensure we don't overwrite silently, since we'll have to check old block for deletion before